src/Couchbase.Lite.Tests.Shared/ReplicationTest.cs

```diff
@@ -70,9 +70,35 @@ public ReplicatorTest()
             ReopenDB();
             Database.Delete("otherdb", Directory);
             _otherDB = OpenDB("otherdb");
+            Database.SetLogLevel(LogDomain.Replicator, LogLevel.Verbose);
         }
 
-        #if COUCHBASE_ENTERPRISE
+        [Fact]
+        public async Task TestCloseServerWhenReplicatorExistsAsync()
+        {
+            var targetEndpoint = new URLEndpoint(new Uri("ws://192.168.0.11:4984/app"));
+            var config = new ReplicatorConfiguration(Db, targetEndpoint);
+            using (var repl = new Replicator(config))
+            {
+                repl.Start();
+                var count = 0;
+
+                this.Invoking(x => ReopenDB())
+                    .ShouldThrow<CouchbaseLiteException>(
+                        "because the database cannot be closed while replication is running");
+            
+                while (count++ > 20 && repl.Status.Activity != ReplicatorActivityLevel.Stopped)
+                {
+                    WriteLine($"Replication status still {repl.Status.Activity}, waiting for stopped...");
+                    await Task.Delay(500);
+                }
+
+                if(repl.Status.Activity == ReplicatorActivityLevel.Stopped)
+                    WriteLine($"Replication status still {repl.Status.Activity}, stopped");
```

Comment:
This should be an assertion, not just a WriteLine.  The test should fail if the replication does not close.  Note that in the situation we are going to cover, retries will be attempted so it will take quite a while for the replication to stop.  I'll catch up with you about that offline.